src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -1656,7 +1656,7 @@ class RGWBucketMetadataHandler : public RGWBucketMetadataHandlerBase {
 class RGWMetadataHandlerPut_Bucket : public RGWMetadataHandlerPut_SObj
 {
   RGWBucketMetadataHandler *bhandler;
-  RGWBucketEntryMetadataObject *obj;
+  RGWBucketEntryMetadataObject *obj = nullptr;
```

Comment:
are you sure this is the correct behavior?
maybe the right fix is to pass `_obj` to the ctor of `RGWMetadataHandlerPut_SObj`?

Reply:
so, why do we pass it to the `RGWMetadataHandlerPut_SObj` constructor?

Reply:
ok. but this is still an issue.
we pass `nullptr` to `RGWMetadataHandler_GenericMetaBE::Put::Put` but store the object pointer to `RGWMetadataHandlerPut_Bucket`.
we should probably try to ask what is the right fix.

Reply:
@kalebskeithley this was done as part of https://github.com/ceph/ceph/commit/5ba9842b92562e33b3770c02d4e62db63383c3da
should we pass `_obj` to `RGWMetadataHandlerPut_SObj` ?

Reply:
@vedanshbhartia could you please fix as discussed?